web-wallet/src/lib/contracts/areValidSettings.js

```diff
@@ -0,0 +1,27 @@
+import { gasStore } from "$lib/stores";
+import { isType } from "lamb";
+import { get } from "svelte/store";
+
+const gasLimits = get(gasStore);
```

Comment:
This way we are assuming that the gas store never changes. Is it a safe assumption?

How can we test it? Can we test that the `gasStore` is read only here?

If we can I'm ok with the assumption, otherwise, probably, to avoid introducing bugs in the future, it's better to read the store inside the function and have a test to check that the function adapts to eventual changes in the store.

What do you think?

N.B.: other tests should be added as well as there are none right now.

Reply:
The only time gasStore changes is when the ENVs are updated in the DO app. So i could say that the store never changes until a new app build is trigger in DO.

Reply:
Yes but as this is a pre-condition for this function to give reliable results, we need to test it.

Reply:
(I mean that such pre-condition can change in the lifecycle of an application, and without a test to ensure it still stands we risk to introduce bugs)